unityfiles/Assets/Scripts/GUI/ButtonChangeLevelState.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+/// <summary>
+/// Button handler for changing level state.
+/// It requires to be on an game object with Button script applied
+/// </summary>
+[RequireComponent(typeof(Button))]
+public class ButtonChangeLevelState : MonoBehaviour
+{
+
+    [SerializeField]
+    private LEVEL_STATE levelState;
+    
+    /// <summary>
+    /// The button that triggers the click event
+    /// </summary>
+    private Button triggerButton;
+
+    private void Awake() {
+        this.triggerButton = GetComponent<Button>();
+        this.triggerButton.onClick.AddListener(ChangeState);
+    }
+
+    private void OnDestroy() {
+        this.triggerButton.onClick.RemoveListener(ChangeState);
+    }
+
+    private void ChangeState(){
+        LevelManager.Instance.ChangeLevelState(this.levelState);
+    }
+}
```

Comment:
Denne klassen har mulighet for å tappe seg inn på LevelState change; kan brukes på GUI knapper.
Bare for infor :D